hearts/api/socket_events.py

```diff
@@ -90,13 +90,20 @@ def sid_from_player(room, player):
             return user_data['socket_id']
 
 
-def emit_game_updates(room, game):
+def emit_game_updates(room, game, player_id=None):
     '''
     Emit a serialized view of the game to each player in the room.
+    Using the optional player_id argument will update only for that player.
     '''
-    for user_info in room['users']:
-        serialized_for_player = game.serialize(for_player=Player(user_info['username']))
-        io.emit('game_update', serialized_for_player, room=user_info['socket_id'])
+
+    if player_id:
+        for_player = player_from_sid(room, player_id)
+        serialized_for_player = game.serialize(for_player)
```

Comment:
Nitpick: prefer to use the kwarg explicitly, `for_player=for_player`

Reason is that if the arguments of that function change later, you won't be implicitly relying on the order to determine which value goes to which argument.